### ISSUE TITLE
Problem with streaming queries

### ISSUE THREAD

**Issue by tgrandje** [requests-cache/requests-cache#878]
### Problem with streaming queries
Queries do not seem to stream appropriately. It seems the whole response is awaited by the CachedSession before releasing the handle (the example in the doc is not appropriate as it is very quick).

### Expected behavior
Chunks should be parsed as soon as they are gathered.

### Steps to reproduce the behavior
```
import requests
import requests_cache

url = "https://wxs.ign.fr/859x8t863h6a09o9o6fy4v60/telechargement/inspire/BDTOPO-FRANCE-ADMIN-SSDOUBLON-PACK_17.1$BDTOPO_2-2_ADMINISTRATIF_SHP_WGS84G_FRA_2017-01-01/file/BDTOPO_2-2_ADMINISTRATIF_SHP_WGS84G_FRA_2017-01-01.7z"

s = requests.Session()
r = s.get(url, stream=True)
print(r.status_code)
for chunk in r.iter_content(chunk_size=1024):
    print(chunk[:10])
    break  # will break asap

requests_cache.install_cache()
r = s.get(url, stream=True)
print(r.status_code)
for chunk in r.iter_content(chunk_size=1024):
    print(chunk[:10])
    break # will still do

# Lets' use a new cache
s = requests_cache.CachedSession("dummy")
r = s.get(url, stream=True)
print(r.status_code) # will not even be printed until the end of the request
for chunk in r.iter_content(chunk_size=1024):
    print(chunk[:10])
    break
```

### Workarounds
It seems the install_cache is enough to circumvent this behaviour. I checked the behaviour of requests_cache and can confirm that the `stream` argument is correctly passed to requests.

### Environment
- requests-cache version: `1.1.0`
- Python version: `3.9`
- Platform: Windows 10 pro

**Comment by JWCook** [comment]
You're correct, the current level of support for streaming requests is making sure the stream can be played back correctly when returned from the cache (i.e., so chunking behavior is the same between original and cached responses). During the initial request, the entire response contents are read and cached before being returned to the user. I'm not able to reproduce any difference in behavior with `install_cache()`, though.

There would be a few different ways to approach caching a streaming response only after it reaches the end of the stream, but I can't think of any particularly clean solutions right now. I will need to give that some more thought. Meanwhile, I will try to at least come up with a workaround you can use.

**Comment by nschloe** [comment]
Not sure if related: The following stream-to-extract requests succeeds with `requests`, but fails with `requests-cache`:

```python
import requests
import tarfile

import requests_cache

url = "https://arxiv.org/e-print/1102.00510v1"

# works:
# r = requests.get(url, timeout=10, stream=True)

# doesn't:
cached_session = requests_cache.CachedSession()
r = cached_session.get(url, timeout=10, stream=True)


with tarfile.open(fileobj=r.raw) as tar:
    tar.extractall(path="/tmp")
```

The error message is

```
requests.exceptions.ChunkedEncodingError: ('Connection broken: IncompleteRead(50440 bytes read, -25220 more expected)', IncompleteRead(50440 bytes read, -25220 more expected))
```

**Comment by JWCook** [comment]
Thanks for the example. I'm guessing `Content-Length` isn't being set correctly. I'll look into it!

**Comment by JWCook** [comment]
I don't think that's related, though. Could you create a separate issue for that, please?
